### PR TITLE
refactor(react-scheduler): use getter instead of template to pass parameters to appointments

### PR DESCRIPTION
**Pull request by AryamnovEugeniy**

### CODE REVIEW

File: packages/dx-scheduler-core/src/utils.ts [hunk 1]
```diff
@@ -320,14 +323,17 @@ export const calculateRectByDateIntervals: CalculateRectByDateIntervalsFn = (
   const { growDirection, multiline } = type;
   const isHorizontal = growDirection === HORIZONTAL_TYPE;
 
-  const sorted = sortAppointments(intervals);
-  const grouped = findOverlappedAppointments(sorted as AppointmentMoment[], isHorizontal);
+  const sorted = intervals.map(intervalGroup => sortAppointments(intervalGroup));
```
Comment (MaximKudriavtsev) [inline packages/dx-scheduler-core/src/utils.ts]:
```js
const sorted = intervals.map(intervalGroup => sortAppointments(intervalGroup));
// same with
const sorted = intervals.map(sortAppointments);
```

File: packages/dx-react-scheduler/src/plugins/month-view.tsx [hunk 2]
```diff
@@ -1,26 +1,20 @@
 import * as React from 'react';
 import { Plugin, PluginComponents } from '@devexpress/dx-react-core';
-import { monthCellsData, horizontalTimeTableRects } from '@devexpress/dx-scheduler-core';
+import { monthCellsData, calculateMonthDateIntervals as calculateMonthDateIntervalsCore } from '@devexpress/dx-scheduler-core';
```
Comment (ushkal) [inline packages/dx-react-scheduler/src/plugins/month-view.tsx]: We don't need to rename it to ...Core

**Comment by AryamnovEugeniy** [comment]
FPR

File: packages/dx-react-scheduler/src/plugins/resources.tsx [hunk 3]
```diff
@@ -1,14 +1,25 @@
...
 import { ResourcesProps } from '../types/resources/resources.types';
-import { Appointments } from '../types';
+import { memoize } from '@devexpress/dx-core';
 
 const pluginDependencies = [
   { name: 'Appointments' },
 ];
 
-const ResourcesBase: React.SFC<ResourcesProps> = ({
+const addResourcesToTimeTableAppointments = memoize(({
```
Comment (MaximKudriavtsev) [inline packages/dx-react-scheduler/src/plugins/resources.tsx]: We should not use memoization in the global context

**Pull request merged**